python_modules/dagster/dagster/_core/definitions/observable_asset.py

```diff
@@ -6,12 +6,77 @@
     AssetExecutionType,
     AssetSpec,
 )
+from dagster._core.definitions.assets import AssetsDefinition
 from dagster._core.definitions.decorators.asset_decorator import asset, multi_asset
 from dagster._core.definitions.source_asset import SourceAsset
 from dagster._core.errors import DagsterInvariantViolationError
 
 
-def create_unexecutable_observable_assets_def(specs: Sequence[AssetSpec]):
+def external_assets_def_from_specs(specs: Sequence[AssetSpec]) -> AssetsDefinition:
+    """Create an external assets definition from a sequence of asset specs.
```

Comment:
what's a use case where you would create an external asset from multiple `AssetSpecs`?

Reply:
as an aside, doing this 
```python
a1 = AssetSpec(key="table_1")
a2 = AssetSpec(key="table_2")
a3 = AssetSpec(key="table_3", deps=[a1])

external_assets = create_unexecutable_observable_assets_def([a1, a2, a3]) 
```

causes errors when running the UI 
```
dagster._core.errors.DagsterInvariantViolationError: Expected Tuple annotation for multiple outputs, but received non-tuple annotation.
```

Reply:
I might not fully understand, but the `assets` in the fn name isn't my concern, it's that you can pass a list of `AssetSpec`s. I don't really get why you would ever pass a list, so I'm wondering why it can't be 
```python
def external_assets_def_from_spec(spec: AssetSpec) -> AssetsDefinition:
```

Reply:
This is more about convenience than anything. People will construct N asset specs to represent the graph and I want them to be able to coerce them into a form that can be threaded to `Definitions` in a single fn call.

Reply:
ok that makes sense. I will note that just with the doc bloc and function signature, I was pretty unsure by what would happen if i passed a list of AssetSpecs. And it does still seem to throw an error if you try to pass a list (but that's fixable). I think the initial confusion can be remedied by addressing what happens when you pass multiple `AssetSpecs` in the doc block, or by generating an `AssetsDefinition` per spec. While the latter is maybe less efficient, it fits my mental model of what I would expect this function to do

Reply:
I'm totally cool with creating multiple assets_def. @sryza is that what you would prefer as well?

Reply:
Yeah that feels more grokkable to me. And I agree that it's ergonomic for it to take a list, rather than requiring the user to write a comprehension.

Reply:
Cool. Makes sense to me.

Reply:
Actually this was a very good. Error message if you somehow get to the user-defined execute function will be much more clear.